src/infer_request.cc

```diff
@@ -1269,47 +1274,89 @@ InferenceRequest::ValidateBytesInputs(
     const std::string& input_id, const Input& input,
     int64_t* const expected_byte_size) const
 {
+  *expected_byte_size = 0;
   const auto& input_dims = input.ShapeWithBatchDim();
+
   int64_t element_count = triton::common::GetElementCount(input_dims);
   int64_t element_idx = 0;
-  *expected_byte_size = 0;
-  for (size_t i = 0; i < input.Data()->BufferCount(); ++i) {
-    size_t content_byte_size;
-    TRITONSERVER_MemoryType content_memory_type;
-    int64_t content_memory_id;
-    const char* content = input.Data()->BufferAt(
-        i, &content_byte_size, &content_memory_type, &content_memory_id);
-
-    while (content_byte_size >= sizeof(uint32_t)) {
-      if (element_idx >= element_count) {
-        return Status(
-            Status::Code::INVALID_ARG,
-            LogRequest() + "unexpected number of string elements " +
-                std::to_string(element_idx + 1) + " for inference input '" +
-                input_id + "', expecting " + std::to_string(element_count));
+  size_t remaining_element_size = 0;
+
+  size_t buffer_idx = 0;
+  const size_t buffer_count = input.Data()->BufferCount();
+
+  const char* buffer = nullptr;
+  size_t remaining_buffer_size = 0;
+  TRITONSERVER_MemoryType buffer_memory_type;
+  int64_t buffer_memory_id;
+
+  // Validate elements until all buffers have been fully processed.
+  while (remaining_buffer_size || buffer_idx < buffer_count) {
+    // Get the next buffer if not currently processing one.
+    if (!remaining_buffer_size) {
+      // Reset remaining buffer size and pointers for next buffer.
+      buffer = input.Data()->BufferAt(
+          buffer_idx++, &remaining_buffer_size, &buffer_memory_type,
+          &buffer_memory_id);
+      *expected_byte_size += remaining_buffer_size;
+
+      // FIXME: Skip GPU buffers for now, return an expected_byte_size of -1 as
+      // a signal to skip validation.
+      if (buffer_memory_type == TRITONSERVER_MEMORY_GPU) {
+        LOG_WARNING << LogRequest()
+                    << "Validation of GPU byte size buffers is not implemented "
+                       "yet, skipping validation for input: "
```

Comment:
I can update the message, it's probably best-effort on a per backend basis rather than strictly validated.

Were you working on updating the GPU buffer validation in core? What was your latest status regarding that? Feel free to slack.

Reply:
Slacked you.